app/assets/javascripts/discourse/app/lib/swipe-events.js

```diff
@@ -1,6 +1,52 @@
 import { isTesting } from "discourse-common/config/environment";
 import { bind } from "discourse-common/utils/decorators";
 
+// common max animation time in ms for swipe events for swipe end
+// prefers reduced motion and tests return 0
+export function getMaxAnimationTimeMs(durationMs = MAX_ANIMATION_TIME) {
+  if (
+    isTesting() ||
+    window.matchMedia("(prefers-reduced-motion: reduce)").matches
+  ) {
+    return 0;
+  }
+  return Math.min(durationMs, MAX_ANIMATION_TIME);
+}
+
+//functions to calculate if a swipe should close
+//based on origin of right, left, top, bottom
+// menu should close after a swipe either:
+// if a user moved the panel closed past a threshold and away and is NOT swiping back open
+// if a user swiped to close fast enough regardless of distance
+export function shouldCloseMenu(e, origin) {
+  if (origin === "right") {
+    return (
+      (e.deltaX > SWIPE_DISTANCE_THRESHOLD &&
+        e.velocityX > -SWIPE_VELOCITY_THRESHOLD) ||
+      e.velocityX > 0
+    );
+  } else if (origin === "left") {
+    return (
+      (e.deltaX < -SWIPE_DISTANCE_THRESHOLD &&
+        e.velocityX < SWIPE_VELOCITY_THRESHOLD) ||
+      e.velocityX < 0
+    );
+  } else if (origin === "bottom") {
+    return (
+      (e.deltaY > SWIPE_DISTANCE_THRESHOLD &&
+        e.velocityY > -SWIPE_VELOCITY_THRESHOLD) ||
+      e.velocityY > 0
+    );
+  } else if (origin === "top") {
+    return (
+      (e.deltaY < -SWIPE_DISTANCE_THRESHOLD &&
+        e.velocityY < SWIPE_VELOCITY_THRESHOLD) ||
+      e.velocityY < 0
+    );
+  }
+  return false;
+}
+
```

Comment:
Not sure swipe-events is the right home for this, but it's already better than in the events logic where it was before